02-household-queries/ingest.py

```diff
@@ -0,0 +1,51 @@
+from langchain_google_genai import GoogleGenerativeAIEmbeddings
+from bs4 import BeautifulSoup
+from langchain_community.document_loaders import PDFMinerLoader
+from langchain.docstore.document import Document
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+from langchain_community.vectorstores import Chroma
+import json
+import dotenv
+
+dotenv.load_dotenv()
+
+# split text into chunks
+def get_text_chunks_langchain(text, source):
+    text_splitter = RecursiveCharacterTextSplitter(chunk_size=1000, chunk_overlap=500)
+    texts = text_splitter.split_text(text)
+    docs = [Document(page_content=t, metadata={"source":source}) for t in texts]
+    return docs
+
+# Chunk the pdf and load into vector db
+def add_pdf_to_vector_db(file_path, chunk_size=500, chunk_overlap=100):
+    # PDFMinerLoader only gives metadata when extract_images=True due to default using lazy_loader
+    loader = PDFMinerLoader(file_path, extract_images=True)
+    text_splitter = RecursiveCharacterTextSplitter(chunk_size=chunk_size, chunk_overlap=chunk_overlap)
+    pdf_pages = loader.load_and_split(text_splitter)
+    print("Loading PDF chunks into vector db")
+    vectordb.add_documents(documents=pdf_pages)
+
+# Chunk the json data and load into vector db
+def add_json_html_data_to_vector_db(file_path, content_key, index_key):
+    data_file = open(file_path)
+    json_data = json.load(data_file)
+
+    for content in json_data:
+        soup = BeautifulSoup(content[content_key], "html.parser")
+        text = soup.get_text(separator='\n', strip=True)
+        chunks = get_text_chunks_langchain(text, content[index_key])
+        print(f"Loading Document {content[index_key]} chunk into vector db")
+        vectordb.add_documents(documents=chunks)
+
+embeddings = GoogleGenerativeAIEmbeddings(model="models/embedding-001")
```

Comment:
In another PR, I'd like to have the embedding method more easily adjustable so that I can 1. use a free embedding that doesn't have usage limits or cost and 2. experiment with different embedding methods.
The simplest implementation is to wrap this in a function that is used by `ingest.py` and `retrieval.py`.

Reply:
Will follow up in https://navalabs.atlassian.net/browse/DST-148